var/spack/repos/builtin/packages/libceed/package.py

```diff
@@ -31,8 +33,10 @@ class Libceed(Package):
     conflicts('+magma', when='@:0.5')
 
     depends_on('cuda', when='+cuda')
+    depends_on('hip@3.8.0', when='@0.7+hip')
```

Comment:
May be better to split this into a conflict:
```
    conflicts('+hip', when='@:0.6')
```
and allow newer versions of hip with:
```
    depends_on('hip@3.8.0:', when='+hip')
```

Reply:
That's not very future-proof -- HIP is moving fast and there may well be issues with newer versions. We can add the conflict.

Reply:
I added the conflict, but not the open-ended HIP version.

Reply:
As it is now, `libceed@develop+hip` will not depend on `hip`. Maybe switch from `'when=@0.7+hip'` to just `'when=+hip'` -- the conflict will take care of the libceed version.

Reply:
Seems like more complicated logic to put in version constraints after future releases, so I added this using `depends_on`.